pkg/pillar/types/zedroutertypes.go

```diff
@@ -128,6 +128,50 @@ func (status AppNetworkStatus) VerifyFilename(fileName string) bool {
 			fileName, expect)
 	}
 	return ret
+
+}
+
+// PerInterfaceErrorMap - Used to return per-interface errors.
+//  ifName is usually used as the key
+type PerInterfaceErrorMap struct {
+	IntfErrorMap map[string]ErrorInfo
+}
+
+// AddError - Add an error into PerInterfaceErrorMap. If en entry already
+//  exists for the interface, it is appended
+func (intfMap *PerInterfaceErrorMap) AddError(ifName string,
+	err error, source string) {
+	errInfo, ok := intfMap.IntfErrorMap[ifName]
+	if ok {
+		errInfo.Error = errInfo.Error + " \n " + err.Error()
+		if errInfo.ErrorSource == "" {
+			errInfo.ErrorSource = source
+		} else {
+			errInfo.ErrorSource += ", " + source
+		}
```

Comment:
As I said  above, this is a misunderstanding of what ErrorSource is. Further appending them together doesn't add any value the way you incorrectly use it. I don't think it even gets used in your diff. But it doesn't matter since it shouldn't even be there.

Reply:
Well.. The conditions are:

1) Can an interface have multiple Errors?
2) Are each of them coming from different places?

If yes - this is the best I could imagine them - without too much complexity.
the errors are separated by "\n" and the sources are "," separated. Since we get to see both, we can quickly figure out where the errors are coming from. I had the option of:

1) Getting rid of the source field
2) Overwriting source field
3) Concatenating them as I did here
4) Do something more complex..

Took 3 as the choice because it is still useful to me. With these proper abstractions in place, we can evolve them more easily as and when we need.

Reply:
Did you miss out discussion in slack about errorSource? You should not reuse the zedmanager's ErrorInfo. Define something fit for purpose like:
// IntfErrInfo contains the result of testing/verifying one interface in
// a DevicePortConfig
// LastError is set if LastFailed is after LastSucceeded
// If neither timestamp is set it means the interface was
// not tested as part of this DevicePortConfig
type IntfErrInfo struct {
	LastSucceeded time.Time
	LastFailed    time.Time
	LastError     error
}

And if it turns out that you need to or want to represent this as a single timestamp+string in the EVE API, then you can do that processing when sending to the controller; the two timestamps makes it easier to observe nim on the device when debugging issues.

And this PR still has all of the unrelated cleanup which I think you said you were moving out.
So I can't review this.